docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+
+	if err != nil {
+		fmt.Fprintf(w, "failed to connect: %s", err.Error())
+		return
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			fmt.Fprintf(w, "failed to disconnect: %s", err.Error())
+			return
+		}
+	}()
+
+	// Ping the primary
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		fmt.Fprintf(w, "unable to connect to the server: %s", err.Error())
+		return
+	}
+
+	fmt.Fprintf(w, "Successfully connected and pinged.")
+}
+```
+We will be using the mongodb client library, so let's update the go.mod with this dependency by running the following command: 
+```shell
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+### Run the application on the cluster
+Run this application on the cluster with `odo dev`.
+```shell
+odo dev
+```
+```shell
+$ odo dev
+  __
+ /  \__     Developing using the my-go-app Devfile
+ \__/  \    Namespace: my-percona-server-mongodb-operator
+ /  \__/    odo version: v3.0.0-rc1
+ \__/
+
+↪ Deploying to the cluster in developer mode
+ •  Waiting for Kubernetes resources  ...
+ ⚠  Pod is Pending
+ ✓  Pod is Running
+ ✓  Syncing files into the container [152ms]
+ ✓  Building your application in container on cluster (command: build) [15s]
+ •  Executing the application (command: run)  ...
+ -  Forwarding from 127.0.0.1:40001 -> 8080
+
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/.odo changed
+ •  Waiting for Kubernetes resources  ...
+ ✓  Syncing files into the container [1ms]
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+```
+
+### Check the connection
+
+Once the application is running, try calling the URL and check the response.
+```shell
+curl 127.0.0.1:40001
+```
+```shell
+$ curl 127.0.0.1:40001
+failed to connect: error validating uri: username required if URI contains user info
+```
+
+This response is expected because we have not yet exposed the connection information to our cluster environment and hence have failed to connect to the mongodb service.
+
+### Connect the application to the mongodb service
+Let's now connect our application to the mongodb service with `odo add binding`.
+
+From a new terminal, run the following command that will add necessary data to devfile.yaml:
+```shell
+odo add binding \
+  --service mongodb-instance/PerconaServerMongoDB \
+  --name my-go-app-mongodb-instance \
+  --bind-as-files=false
+```
+```shell
+$ odo add binding --service mongodb-instance/PerconaServerMongoDB --name my-go-app-mongodb-instance --bind-as-files=false
+ ✓  Successfully added the binding to the devfile.
+Run `odo dev` to create it on the cluster.
+```
+
+:::note
+`--binding-as-files=false` because our code logic relies on obtaining environment variables from the system instead of reading data from files.
```

Comment:
I feel like this part should even have it's own section to explain what's happening rather than just a note. Pretty important that the user does not miss this.

Reply:
I've added a link to the add-binding.md reference where this has been explained.